### PR TITLE
add change-branch function

### PR DESCRIPTION
**Pull request by poupotte**

### CODE REVIEW

File: server/controllers/applications.coffee [hunk 1]
```diff
@@ -69,6 +69,46 @@ module.exports.install = (req, res, next) ->
...
+
+    # Stop app if it started
+    controller.stop name, (err, result) ->
+        if err? and err.toString() is 'Error: Cannot stop an application not started'
+            # If application is not started, don't restart it after branch change
+            started = false
+        else if err?
+            # If stop function send another error, stop process
+            log.error err.toString()
+            return sendError res, err, 400
```
Comment (frankrousseau) [inline server/controllers/applications.coffee]: Please, dont use return in a middle of a function.

File: server/controllers/routes.coffee [hunk 2]
```diff
@@ -17,6 +17,11 @@ module.exports =
             applications.install
         ]
 
+    "apps/:name/change-branch/:branch":post: [
```
Comment (frankrousseau) [inline server/controllers/routes.coffee]: Could you put post keyword on newline?

File: server/lib/git.coffee [hunk 3]
```diff
@@ -66,11 +66,13 @@ module.exports.init = (app, callback) ->
...
                     if not gitVersion? or \
                        compareVersions("1.7.10", gitVersion[1]) is 1
-                        commands.push "git clone #{url} #{app.name} && " + \
-                                      "cd #{app.dir} && " + \
-                                      "git branch #{branch} origin/#{branch} && " + \
-                                      "git checkout #{branch} && " + \
-                                      "git submodule update --init --recursive"
+                        command = "git clone #{url} #{app.name} && " + \
+                                  "cd #{app.dir} && "
+                        if branch isnt 'master'
```
Comment (frankrousseau) [inline server/lib/git.coffee]: Not sure about that one. If the current branch is development we should be able to come back on the master branch.

File: server/lib/controller.coffee [hunk 4]
```diff
@@ -230,6 +230,34 @@ module.exports.start = (manifest, callback) ->
...
 
 ###
+    Change aplication branch
+        * Git checkout
+        * Install dependencies
+###
+module.exports.changeBranch = (connection, manifest, newBranch, callback) ->
+    # Git checkout
+    app = new App manifest
+    app = app.app
```
Comment (frankrousseau) [inline server/lib/controller.coffee]: Are you sure about that assignation?

File: server/lib/controller.coffee [hunk 5]
```diff
@@ -230,6 +230,34 @@ module.exports.start = (manifest, callback) ->
...
+    log.info "#{app.name}:git checkout"
+    type['git'].changeBranch app, newBranch, (err) ->
+        if err?
+            # Error on source retrieval : code 2-
```
Comment (frankrousseau) [inline server/lib/controller.coffee]: In english there is no blank space before `:`.

File: server/lib/git.coffee [hunk 6]
```diff
@@ -113,3 +115,20 @@ module.exports.update = (app, callback) ->
...
     executeUntilEmpty commands, config, callback
+
+###
+    Change branch of <app>
+###
+module.exports.changeBranch = (app, newBranch, callback) ->
+
+    # Setup the git commands to be executed
+    commands = [
+        "git fetch origin #{newBranch}:#{newBranch} && " + \
```
Comment (frankrousseau) [inline server/lib/git.coffee]: why do you use `&&` there. The `executeUntilEmpty` function runs all command in the array.

File: server/lib/git.coffee (hunk 3 above)
Comment (poupotte) [inline server/lib/git.coffee]: Function init, is only use for installation.

**Comment by frankrousseau** [comment]
Thank you @poupotte

**Pull request merged**